### DISCUSSION TITLE
Get id back from combobox

### DISCUSSION THREAD

**Discussion by CordlessWool** [carbon-design-system/carbon-components-svelte#1921] [General]
Hello,

I struggle a little bit with the combobox. I want to get the id back, but by sending the form I only get the value. From the current implementation it is clear why this happens, but is this also the idea behind the combobox? It looks like, that it is implemented the same way in the originial carbon design system?

**Comment by theetrain** [comment]
Hello, since the ComboBox represents an `<input />` with some enhanced keyboard functionality, it will pass the selected item's `text` value as its FormData `value` when submitted within a `<form>`.

Demo: https://svelte.dev/repl/88d146568c32490b9ba3aecfdd761d0b?version=4.2.12

I can see how this may be undesirable since `<select>` would submit the selected `<option>`'s `value` and not the option's label. Perhaps having the selected ComboBox item's `id` be the submitted `value` is better; let us know what you think.

**Reply by pekeler** [reply]
Hi, I want to make a to-one-relationship editable. The number of possible entities is too large for a select, so the idea is to use a ComboBox instead and dynamically update the shown options based on that the user has entered in the input. The entity's name property should be the label, and the entity's primary-key should be the value. The entity's name is not always guaranteed to be unique which makes it necessary to use the entity's primary-key as value.

Or putting it more generic: if the number of options becomes too large for a select, one should use a ComboBox. This makes it necessary that the ComboBox works similar to the select.

That is, unless I'm misunderstanding the purpose of the ComboBox and perhaps Carbon wants me to use something else in this scenario.
